aggregator/aggregator.go

```diff
@@ -148,7 +148,8 @@ type aggregator struct {
 	instanceID    string
 	shardFn       ShardFn
 	checkInterval time.Duration
-	handler       Handler
+	flushManager  FlushManager
```

Comment:
I don't think it's too big an issue, since the aggregator is the owner of these. If golang has destructors and unique_ptr objects, this is like passing in an unique_ptr into the aggregator and the aggregator destructor destroys both when they go out of scope. Unfortunately we don't have destructors or unique_ptr objects, so we have to do it inside `Close`.